pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -223,16 +223,16 @@ def save_checkpoint(self, trainer, pl_module):
         monitor_candidates = self._monitor_candidates(trainer)
 
         # ie: path/val_loss=0.5.ckpt
-        filepath = self._get_metric_interpolated_filepath_name(epoch, monitor_candidates)
+        filepath = self._get_metric_interpolated_filepath_name(epoch, global_step, monitor_candidates)
 
         # callback supports multiple simultaneous modes
         # here we call each mode sequentially
         # Mode 1: save all checkpoints OR only the top k
         if self.save_top_k:
-            self._save_top_k_checkpoints(monitor_candidates, trainer, pl_module, epoch, filepath)
+            self._save_top_k_checkpoints(monitor_candidates, trainer, pl_module, epoch, global_step, filepath)
 
```

Comment:
instead of adding to the `_save_top_k_checkpoints ` interface here, you could add `epoch` and `global_step` to the `monitored_candidates` dict so you don't have to explicitly pass them around everywhere. i am slightly concerned that `_save_top_k_checkpoints` will gradually accumulate more and more arguments over time.

Reply:
ok, sounds good. 
I did an intermediate step and remove it from `_save_last_checkpoint`, `_get_metric_interpolated_filepath_name`, and `_save_top_k_checkpoints`. 

We can pass it down even further into `format_checkpoint_name`, but then I would have to change many tests. We could leave this to a new refactor PR?